firestore/client.go

```diff
@@ -234,7 +257,9 @@ func (c *Client) getAll(ctx context.Context, docRefs []*DocumentRef, tid []byte,
 		req.ConsistencySelector = &pb.BatchGetDocumentsRequest_Transaction{Transaction: tid}
 	}
 
-	streamClient, err := c.c.BatchGetDocuments(withResourceHeader(ctx, req.Database), req)
+	batchGetDocsCtx := withResourceHeader(ctx, req.Database)
+	batchGetDocsCtx = withRequestParamsHeader(batchGetDocsCtx, vkit.ReqParamsHeaderVal(c.path()))
```

Comment:
I don't have much context on this -- why are we updating the context here?

Reply:
Firestore service requires headers.
Currently, only 'google-cloud-resource-prefix' metadata header gets sent with all the requests to Firestore service. The value in this header indicates the resource being operated on.

But while using named databases i.e. database other than default database, 'x-goog-request-params' dynamic routing header is required. This is the change at this line.